GitUp/Application/AppDelegate.m

```diff
@@ -68,6 +79,113 @@ - (BOOL)canBecomeKeyWindow {
 
 @end
 
+@interface GIWelcomeWindow__DraggingInfoExtractor : NSObject
+- (BOOL)canDragItem:(id<NSDraggingInfo>)sender;
+- (NSArray <NSURL *>*)draggingItems:(id<NSDraggingInfo>)sender;
+@end
+
+@implementation GIWelcomeWindow__DraggingInfoExtractor
+- (BOOL)canDragItem:(id<NSDraggingInfo>)sender {
+  return [self draggingItems:sender].count > 0;
+}
+- (NSArray <NSURL *>*)draggingItems:(id<NSDraggingInfo>)sender {
+  return [[sender draggingPasteboard] readObjectsForClasses:@[NSURL.class] options:nil];
+}
+@end
+
+@interface GIWelcomeWindow__DestinationView : NSView
+@property (weak, nonatomic, readwrite) AppDelegate *appDelegate;
+@property (strong, nonatomic, readwrite) GIWelcomeWindow__DraggingInfoExtractor *draggingExtractor;
+@property (assign, nonatomic, readwrite) BOOL receivingDrag;
+@property (strong, nonatomic, readwrite) NSView *imageView;
+@end
+
+@interface GIWelcomeWindow__DestinationView (Configurations)
+- (instancetype)configuredWithAppDelegate:(AppDelegate *)appDelegate;
+@end
+
+@implementation GIWelcomeWindow__DestinationView (Configurations)
+- (instancetype)configuredWithAppDelegate:(AppDelegate *)appDelegate {
+  self.appDelegate = appDelegate;
+  return self;
+}
+@end
+
+@interface GIWelcomeWindow__DestinationView (NSDraggingDestination) <NSDraggingDestination>
+@end
+
+@implementation GIWelcomeWindow__DestinationView
+- (void)setup {
+  self.draggingExtractor = [GIWelcomeWindow__DraggingInfoExtractor new];
+  [self registerForDraggedTypes:@[NSURLPboardType]];
+}
+
+- (void)awakeFromNib {
+  [super awakeFromNib];
+  [self setup];
+}
+
+- (void)setReceivingDrag:(BOOL)receivingDrag {
+  _receivingDrag = receivingDrag;
+  [self setNeedsDisplay:YES];
+}
+
+- (void)drawRect:(NSRect)dirtyRect {
+  if (self.receivingDrag) {
+    [NSColor.selectedControlColor set];
+    NSBezierPath *path = [NSBezierPath bezierPathWithRoundedRect:self.bounds xRadius:10 yRadius:10];
+    path.lineWidth = 5;
+    [path stroke];
+  }
+}
+@end
+
+@implementation GIWelcomeWindow__DestinationView (NSDraggingDestination)
```

Comment:
Same as other comment, don't implement these in categories. Categories don't necessarily have the same runtime attributes as Swift extensions.